pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -0,0 +1,332 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traceflow
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/types"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/config"
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+	opsv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/ops/v1alpha1"
+	clientsetversioned "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+	"github.com/vmware-tanzu/antrea/pkg/ovs/ovsconfig"
+)
+
+const (
+	controllerName = "AntreaAgentTraceflowController"
+	// How long to wait before retrying the processing of a traceflow
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+	// Default number of workers processing traceflow request
+	defaultWorkers = 1
+)
+
+// Controller is responsible for setting up Openflow entries and inject traceflow packet into
+// switch for traceflow request.
+type Controller struct {
+	traceflowClient clientsetversioned.Interface
+	ovsBridgeClient ovsconfig.OVSBridgeClient
+	ofClient        openflow.Client
+	interfaceStore  interfacestore.InterfaceStore
+	networkConfig   *config.NetworkConfig
+	nodeConfig      *config.NodeConfig
+	nodeLister      corelisters.NodeLister
+	namespaceLister corelisters.NamespaceLister
+	podLister       corelisters.PodLister
```

Comment:
We made a lot of efforts to not watching all Pods from all Nodes. It will affect a scale cluster, considering 100k pods need to be received by 1k Nodes.

I think we have two options here:
1. Controller calculates the dst IP and fills it in CRD
2. Agent queries the dst Pod on demand

Reply:
Currently we use second method.
podInformer := informerFactory.Core().V1().Pods()
podLister:       podInformer.Lister(),
dstPod, err := c.podLister.Pods(tf.Spec.Destination.Namespace).Get(tf.Spec.Destination.Pod)

Will this method watch all Pods?

Reply:
Changed to clientset:
dstPod, err := c.kubeClient.CoreV1().Pods(tf.Spec.Destination.Namespace).Get(tf.Spec.Destination.Pod, v1.GetOptions{})